event-dispatcher/src/main/java/io/sk8s/event/dispatcher/KafkaConsumerMonitor.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sk8s.event.dispatcher;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+import java.io.Closeable;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.springframework.util.StopWatch;
+
+/**
+ * Tracks current and end offset (and hence lag) for a set of topic/consumerGroup pairs.
+ *
+ * @author Eric Bottard
+ */
+public class KafkaConsumerMonitor implements Closeable {
+
+	private final String host;
+
+	private final Map<String, Consumer<?, ?>> consumersByGroup = new HashMap<>();
+
+	private final String monitoringConsumerGroupID = "monitoring_consumer_" + UUID.randomUUID().toString();
+
+	private volatile Consumer<?,?> logEndOffsetTrackingConsumer;
+
+	private final Set<TopicGroup> tracked = new HashSet<>();
+
+	private final static Collection<TopicPartition> FOR_ALL_PARTITIONS = Collections.emptyList();
```

Comment:
could this just be "PARTITIONS"? I'm not following the meaning of "FOR_ALL_".

Reply:
Using an empty collection instructs the method where it is used to reply for all assigned partitions. Used a constant to make that explicit, apparently I failed...

Reply:
I see. If it's only used in one place, that seems unnecessary (maybe a comment where invoked with the empty collection instead?)